### DISCUSSION TITLE
csv import

### DISCUSSION THREAD

**Discussion by rafikApp** [Budibase/budibase#7301] [Help] [answered]
Hello everyne.
How many rows we can have in a data table? because when I want to import a csv file with 9695 rows I get this error

![Capture](https://user-images.githubusercontent.com/106697208/184863818-242198b2-f0ff-4019-a57c-fe8215d32128.PNG)

**Comment by andz-bb** [comment, marked as answer]
Hey, do you get any errors in the console or network tab in your browser developer tools? You can see these when you press F12 and try the import again. If you get errors there then they should show what is going wrong in a bit more detail to help us work out what the problem is. Cheers

**Reply by rafikApp** [reply]
I get these errors in the console
![Capture](https://user-images.githubusercontent.com/106697208/184989064-11cb53cb-2c37-4ef5-bb71-a1d52640f388.PNG)

**Reply by andz-bb** [reply]
hey @rafikApp, unfortunately you're running into row limits on our cloud hosted free tier. At the minute you can only have up to 4000 rows, but we are launching paid tiers with higher resource limits very soon. In the meantime if you need to work with large datasets you can try Budibase self hosted on your computer. It's pretty quick and easy to set it up [using this guide.](https://docs.budibase.com/docs/budibase-cli-setup)
